assets/helpers/contributions.js

```diff
@@ -21,43 +26,45 @@ export type Amounts = {
   oneOff: Amount,
 };
 
+export type validatedContrib = {
+  amount: number,
+  error: ContribError,
+};
+
 
 // ----- Setup ----- //
 
-const limits = {
+const LIMITS = {
   max: 2000,
   min: {
-    monthly: 5,
-    oneOff: 1,
+    RECURRING: 5,
+    ONE_OFF: 1,
   },
 };
 
+const DEFAULT_AMOUNTS = {
+  RECURRING: 10,
+  ONE_OFF: 50,
+};
 
-// ----- Functions ----- //
 
-function validate(amount: Amount, contrib: Contrib): ?ContribError {
+// ----- Functions ----- //
 
-  if (amount.value === '') {
-    return 'invalidEntry';
-  }
+export default function validate(input: string, contrib: Contrib) {
 
-  const numericAmount = Number(amount.value);
+  let error = null;
+  const numericAmount = Number(input);
 
-  if (isNaN(numericAmount)) {
-    return 'invalidEntry';
-  } else if (numericAmount < limits.min.monthly && contrib === 'RECURRING') {
-    return 'tooLittleRecurring';
-  } else if (numericAmount < limits.min.oneOff && contrib === 'ONE_OFF') {
-    return 'tooLittleOneOff';
-  } else if (numericAmount > limits.max) {
-    return 'tooMuch';
+  if (input === '' || isNaN(numericAmount)) {
+    error = 'invalidEntry';
+  } else if (numericAmount < LIMITS.min[contrib]) {
+    error = contrib === 'RECURRING' ? 'tooLittleRecurring' : 'tooLittleOneOff';
```

Comment:
I think it might make sense for this to return `tooLittle`, and handle the different error message for recurring/monthly elsewhere.